modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelect.scala

```diff
@@ -50,22 +56,28 @@ object MajorityPeerSelect {
 
     def filterPeerList(peers: NonEmptyList[Peer]): F[NonEmptyList[Peer]] =
       peers
-        .traverse(snapshotClient.getLatestOrdinal(_))
-        .map {
-          _.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+        .parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.getLatestOrdinal(_))
+        .map { latestOrdinals =>
+          val (majorityOrdinal, _) = latestOrdinals.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+          majorityOrdinal
+        }
+        .flatMap { majorityOrdinal =>
+          peers.parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.get(majorityOrdinal).run(_).attempt)
```

Comment:
if peer returned latest ordinal higher than the majority ordinal but does not have the majority ordinal, it's an unexpected error and should disqualify that peer from selecting.

Reply:
If a `Peer` returns a `NotFound` for the snapshot given an ordinal, then the query for that `Peer` returns an error in the `F` context. The `attempt` makes that a `Left`, so the following `flatMap` ignores it.